sdk/lib/atl/atlwin.h

```diff
@@ -320,9 +320,37 @@ class CWindow
         int wndCenterHeight = wndCenterRect.bottom - wndCenterRect.top;
         int wndWidth = wndRect.right - wndRect.left;
         int wndHeight = wndRect.bottom - wndRect.top;
+        int xPos = wndCenterRect.left + ((wndCenterWidth - wndWidth + 1) >> 1);
+        int yPos = wndCenterRect.top + ((wndCenterHeight - wndHeight + 1) >> 1);
+
+        if (!(::GetWindowLong(hWndCenter, GWL_STYLE) & WS_CHILD))
+        {
+            MONITORINFO m_MonInfo;
+            m_MonInfo.cbSize = sizeof(MONITORINFO);
+            HMONITOR hMonitor = MonitorFromWindow(hWndCenter, MONITOR_DEFAULTTOPRIMARY);
+            GetMonitorInfo(hMonitor, &m_MonInfo);
+
+            if (xPos + wndWidth > m_MonInfo.rcWork.right)
+            {
+                xPos = m_MonInfo.rcWork.right - wndWidth;
+            }
+            else if (xPos < m_MonInfo.rcWork.left)
+            {
+                xPos = m_MonInfo.rcWork.left;
+            }
+
+            if (yPos + wndHeight > m_MonInfo.rcWork.bottom)
+            {
+                yPos = m_MonInfo.rcWork.bottom - wndHeight;
+            }
+            else if (yPos < m_MonInfo.rcWork.top)
```

Comment:
If you make this `else if` into an `if`, it will compensate for the fact if the windows is higher than the screen,
so that at least the title bar is visible.